content/doc/book/using/distributed-builds.adoc

```diff
@@ -0,0 +1,553 @@
+* Allow port-forwarding from the controller to your agents within the
+firewall. The port-forwarding should be restricted so that only the
+controller with its known IP can connect to agents. With this set up in the
+firewall, as far as Jenkins is concerned it's as if the firewall doesn't
+exist.  If multiple hops are involved, you may wish to investigate how
+to do ssh "jump host" transparently using the ProxyCommand construct.
+In fact,  with a properly configured "jump host" setup, even the controller
+doesn't need to expose itself to the public Internet at all - as long as
+the organization's firewall allows port 22 traffic.
+
+* Use JNLP agents and have agents connect to the controller, not the other
+way around. In this case it's the agents that initiates the connection,
+so it works correctly with the NAT firewall.
+
+Note that in both cases, once the controller is compromised, all your agents
+can be easily compromised (IOW, malicious controller can execute arbitrary
+program on agents), so both set-up leaves much to be desired in terms of
+isolating security breach.
+https://wiki.jenkins.io/display/JENKINS/Build+Publisher+Plugin[Build Publisher Plugin] provides another way of doing this, in more secure
+fashion.
+
+== Running Multiple Agents on the Same Machine
+
+Using a well established virtualization infrastructure such
+as http://en.wikipedia.org/wiki/Kernel-based_Virtual_Machine[Kernel-based Virtual Machine (KVM)], it is quite easy to run multiple agent instances
+on a single physical node.  Such instances can be running various Linux,
+*BSD UNIX, Solaris, Windows.  For Windows, one can have them installed
+as separate Windows services so they can start up on system startup.
+While the correct use of executors largely obviates the need for
+multiple agent instances on the same machine, there are some unique use
+cases to consider:
+
+* You want more configurability between the configured nodes. Say you
+have one node set to be used as much as possible, and the other node to
+be used only when needed.
+* You may have multiple Jenkins controller installations building different
+things, and so this configuration would allow you to have agents for
+more than one controller on the same box. That's right, with Jenkins you
+really can serve two controllers.
+* You may wish to leverage the easiness of starting/stopping/replacing
+virtual machines, perhaps in conjunction with Jenkins plugins such as
+the
+https://wiki.jenkins-ci.org/display/JENKINS/Libvirt+Slaves+Plugin[Libvirt Slaves Plugin].
+* You wish to maximize your hardware investment and utilization, at the
+same time minimizing operating cost (e.g. utility expenses for running
+idling agents).
+
+Follow these steps to get multiple agents working on the same Windows
+box:
+
+* Add the first agent node in Jenkins and give it its own working dir
+(e.g. jenkins-agent-a).
+* Go to the agent page from the agent box and launch by JNLP, then use
```

Comment:
Probably shouldn't document this but I don't know enough about Windows for the service side, the only windows I've used is cloud based agents,

the command from `Run from agent command line:` should be used instead